frontend/public/components/monitoring.tsx

```diff
@@ -1183,265 +1184,264 @@ const Datetime = (props) => {
   );
 };
 
-const durationOffText = '-';
-const durations = [durationOffText, '30m', '1h', '2h', '6h', '12h', '1d', '2d', '1w'];
+const durationOff = '-';
+const durations = [durationOff, '30m', '1h', '2h', '6h', '12h', '1d', '2d', '1w'];
 const durationItems = _.zipObject(durations, durations);
 
-class SilenceForm_ extends React.Component<SilenceFormProps, SilenceFormState> {
-  constructor(props) {
-    super(props);
-
-    const now = new Date();
-    const startsAt = formatDate(now);
-    const endsAt = formatDate(new Date(now.setHours(now.getHours() + 2)));
-    const data = _.defaults(props.defaults, {
-      startsAt,
-      endsAt,
-      matchers: [],
-      createdBy: '',
-      comment: '',
-    });
-    if (_.isEmpty(data.matchers)) {
-      data.matchers.push({ name: '', value: '', isRegex: false });
-    }
-    this.state = { data, duration: '2h', error: undefined, inProgress: false };
-  }
-
-  setField = (path: string, v: any): void => {
-    const data = Object.assign({}, this.state.data);
-    _.set(data, path, v);
-    this.setState({ data });
-  };
+const SilenceForm_: React.FC<SilenceFormProps> = ({ defaults, Info, title }) => {
+  const now = new Date();
 
-  onFieldChange = (path: string): ((e) => void) => {
-    return (e) => this.setField(path, e.target.value);
-  };
+  const [comment, setComment] = React.useState(defaults.comment ?? '');
+  const [createdBy, setCreatedBy] = React.useState(defaults.createdBy ?? '');
+  const [duration, setDuration] = React.useState(defaults.endsAt ? durationOff : '2h');
+  const [endsAt, setEndsAt] = React.useState(
```

Comment:
Can this be `React.useState<date>`?

Reply:
I did consider making it a `Date`, but went with type `string` because we want to store the exact text entered by the user (it's a text input).